AdyenEncryption/CardEncryptor.swift

```diff
@@ -37,161 +21,181 @@ public enum CardEncryptor {
+    ///  maybe because the card public key is an invalid one, or for any other reason.
+    /// - Throws:  `CardEncryptor.Error.invalidEncryptionArguments` when trying to encrypt a card with  card number, securityCode,
+    /// expiryMonth, and expiryYear, all of them are nil.
+    public static func encrypt(expirationMonth: String, with publicKey: String) throws -> String {
+        guard !expirationMonth.isEmpty, expirationMonth.allSatisfy({ $0.isNumber }) else {
+            throw Error.invalidExpiryMonth
+        }
+        let payloadBuilder = CardEncryptedDataBuilder(publicKey: publicKey)
+        return try payloadBuilder.encrypt(expiryMonth: expirationMonth)
+    }
+
+    /// Encrypts enspiration year.
+    ///
+    /// - Parameters:
+    ///   - expiryMonth: The expiration year.
+    ///   - publicKey: The public key to use for encryption (format "Exponent|Modulus").
+    /// - Returns: An encrypted token containing enspiration year data.
+    /// - Throws:  `CardEncryptor.Error.encryptionFailed` if the encryption failed,
+    ///  maybe because the card public key is an invalid one, or for any other reason.
+    /// - Throws:  `CardEncryptor.Error.invalidEncryptionArguments` when trying to encrypt a card with  card number, securityCode,
+    /// expiryMonth, and expiryYear, all of them are nil.
+    public static func encrypt(expirationYear: String, with publicKey: String) throws -> String {
+        guard !expirationYear.isEmpty, expirationYear.allSatisfy({ $0.isNumber }) else {
+            throw Error.invalidExpiryYear
+        }
+        let payloadBuilder = CardEncryptedDataBuilder(publicKey: publicKey)
+        return try payloadBuilder.encrypt(expiryYear: expirationYear)
     }
     
     /// Encrypt BIN.
     /// - Parameters:
     ///   - publicKey: The public key to use for encryption (format "Exponent|Modulus").
     ///   - bin: BIN( Bank Identification number) is the first 6 to 12 digits of PAN.
-    public static func encryptedBin(for bin: String, publicKey: String) throws -> String {
+    /// - Returns: An encrypted token containing BIN data.
+    /// - Throws:  `CardEncryptor.Error.encryptionFailed` if the encryption failed,
+    ///  maybe because the card public key is an invalid one, or for any other reason.
+    /// - Throws:  `CardEncryptor.Error.invalidEncryptionArguments` when trying to encrypt a card with  card number, securityCode,
+    /// expiryMonth, and expiryYear, all of them are nil.
+    public static func encrypt(bin: String, with publicKey: String) throws -> String {
         guard !bin.isEmpty, bin.allSatisfy({ $0.isNumber }) else {
             throw Error.invalidBin
         }
         
-        let payload = try JSONEncoder().encode(Bin(value: bin))
-        do {
-            return try Cryptor(aes: .CCM).encrypt(data: payload, publicKey: publicKey)
-        } catch {
-            throw Error.encryptionFailed
-        }
+        let payloadBuilder = BinEncryptedDataBuilder(publicKey: publicKey)
```

Comment:
same here `EncryptedBinDataBuilder` is a better name IMO.

also the builder pattern is not followed, its just a name :-).